src/app/workflow/workflow.component.html

```diff
@@ -233,72 +233,97 @@ <h3>
+                <div [ngSwitch]="workflow.mode">
+                  <div *ngSwitchCase="WorkflowModel.ModeEnum.STUB">
+                    <!-- For old languages, we use the old files component because it's well tested, for new unknown languages, use the new component -->
+                    <div *ngIf="oldLanguages.includes(workflow.descriptorType); else newComponent" class="p-3">
+                      <ng-container *ngTemplateOutlet="oldComponent"></ng-container>
+                    </div>
+                  </div>
+                  <div *ngSwitchCase="WorkflowModel.ModeEnum.HOSTED">
+                    <ng-container *ngTemplateOutlet="hostedComponent"></ng-container>
+                  </div>
+                  <div *ngSwitchCase="WorkflowModel.ModeEnum.DOCKSTOREYML">
+                    <!-- For old languages, we use the old files component because it's well tested, for new unknown languages, use the new component -->
+                    <div *ngIf="oldLanguages.includes(workflow.descriptorType); else newComponent" class="p-3">
+                      <ng-container *ngTemplateOutlet="oldComponent"></ng-container>
+                    </div>
+                  </div>
+                  <div *ngSwitchCase="WorkflowModel.ModeEnum.FULL">
+                    <!-- For old languages, we use the old files component because it's well tested, for new unknown languages, use the new component -->
+                    <div *ngIf="oldLanguages.includes(workflow.descriptorType); else newComponent" class="p-3">
+                      <ng-container *ngTemplateOutlet="oldComponent"></ng-container>
+                    </div>
+                  </div>
+                  <!-- This also handle SERVICE mode -->
+                  <div *ngSwitchDefault>
+                    <ng-container *ngTemplateOutlet="newComponent"></ng-container>
+                  </div>
                 </div>
-
-                <ng-template #noVersionsFilesTab>
-                  <mat-card class="alert alert-info" role="alert">
-                    <mat-icon>info</mat-icon> This service does not have any versions. Create a release/tag on GitHub with a valid
+              </div>
+              <ng-template #noVersionsFilesTab>
+                <div [ngSwitch]="workflow.mode" class="p-3">
+                  <mat-card *ngSwitchCase="WorkflowModel.ModeEnum.STUB" class="alert alert-info" role="alert">
+                    <mat-icon>info</mat-icon> To see files, please refresh the {{ entryType }}.
+                  </mat-card>
+                  <div *ngSwitchCase="WorkflowModel.ModeEnum.HOSTED">
+                    <ng-container *ngTemplateOutlet="hostedComponent"></ng-container>
+                  </div>
+                  <mat-card *ngSwitchCase="WorkflowModel.ModeEnum.DOCKSTOREYML" class="alert alert-info" role="alert">
+                    <mat-icon>info</mat-icon> This {{ entryType }} does not have any versions. Create a release/tag on GitHub with a valid
```

Comment:
Out of curiosity, when something isn't a stub, how does no versions happen? 

Also in comparison with the first noVersionFilesTab case, does this mean files were found (and displayed) but somehow has no versions? Or is this no files and no versions?

Reply:
Some of these cases are impossible, but showing something instead of nothing.

Reply:
If there's no versions, it's assumed to have no files.